app/utils/users.ts

```diff
@@ -74,7 +60,7 @@ export function useUsersInfinite(token: string) {
  */
 export function useUser(userId?: string, token?: string) {
   const url = `${process.env.NEXT_PUBLIC_CR3_API_DOMAIN}/user/get_user/${userId}`;
-  return useSWR<User>(
+  return useSWR<User | UserAPIError>(
```

Comment:
this is one of those spots where we just gotta take it on faith that the API is going to return this types.

Reply:
I don't know to write this without sounding like a lunatic or bitter -- I'm not --  I am entirely impressed and on board with the stack currently! OK - that said

This API response typing is one of those things that I was so impressed with in the T3 stack with regard to how it used tRPC. The functions on the server are defined as normal ol' typescript and when you call them, almost exactly like you call a normal function in the frontend, client code, you get the result of the function just like normal (even thought it ran on the server), and the whole transaction over the wire is type safe with no extra work. You only have to define types like normal, and then only if they can't be inferred. It's kinda crazy how smooth it is.

Reply:
I would love—love!—to move this API into the nextjs app potentially adding an tRPC. It will have to live in the backlog, but I'll open an issue for it.